templates/CRM/Contribute/Form/Contribution.tpl

```diff
@@ -488,7 +488,7 @@
      });
 
      function showHideCancelInfo(obj) {
-       if (obj.find(":selected").text() == 'Refunded' || obj.find(":selected").text() == 'Cancelled') {
+       if (obj.find(":selected").val() == '7' || obj.find(":selected").val() == '3') {
```

Comment:
I'd strongly recommend updating this to retrieve the value from the name. value is not a reliable key.